typed/typed.go

```diff
@@ -24,16 +24,24 @@ import (
 	"sigs.k8s.io/structured-merge-diff/v4/value"
 )
 
+// ValidationOptions is the list of all the options available when running the validation.
+type ValidationOptions int
```

Comment:
Woudl be better if this was not an int and instead something like a `func (v *validatingObjectWalker)` or something similar so that we aren't bound to very simple options in the future